backend/battling/api/serializers.py

```diff
@@ -10,7 +11,71 @@ class Meta:
         fields = ("email", "id")
 
 
+class PokemonSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Pokemon
+        fields = ("poke_id", "name", "img_url", "attack", "defense", "hp")
+
+
+class TeamSerializer(serializers.ModelSerializer):
+    pokemon_1 = PokemonSerializer()
+    pokemon_2 = PokemonSerializer()
+    pokemon_3 = PokemonSerializer()
+
+    class Meta:
+        model = PokemonTeam
+        fields = ("pokemon_1", "pokemon_2", "pokemon_3")
+
+
 class BattleSerializer(serializers.ModelSerializer):
     class Meta:
         model = Battle
         fields = ("id", "creator", "opponent", "winner")
+
+
+class BattleDetailSerializer(serializers.ModelSerializer):
+    creator = UserSerializer()
+    creator_team = serializers.SerializerMethodField()
+    opponent = UserSerializer()
+    opponent_team = serializers.SerializerMethodField()
```

Comment:
Please replace `creator_team` and `opponent_team` with a single attribute called `teams`